backend/app/Models/EngineeringDocument.php

```
@@ -0,0 +1,115 @@
+
+/**
+ * Engineering CAD document (#179). Associated with one of a fixed set of owner
+ * entities via (entity_type, entity_id) — a short allowlisted key rather than a
+ * FQCN, so it stays stable and matches the REST API and avoids a global morph map
+ * that would change how the existing Attachment model stores its type.
+ *
+ * The file lives on the private disk under a server-generated path. Released
+ * documents are immutable. Soft-deletable (Admin -> Trash); the disk file is
+ * removed when the row is force-deleted.
+ */
+class EngineeringDocument extends Model
```

PHPStan (2.2.6)

[error] Class App\Models\EngineeringDocument extends unknown class Illuminate\Database\Eloquent\Model.
Learn more at https://phpstan.org/user-guide/discovering-symbols

(class.notFound)